File: backend/models/expense.go
package models

import (
	"time"
)

type Expense struct {
	ID 			uint      	`gorm:"primaryKey" json:"id"`
	Amount      float64   	`gorm:"not null" json:"amount"`
	UserID		*uint		`json:"user_id"`
	User        *User		`gorm:"foreignKey:UserID;constraint:OnUpdate:CASCADE,OnDelete:SET NULL;" json:"user,omitempty"`
	Category 	string		`gorm:"not null" json:"category"`
	Description string    	`json:"description"`
	Date  		time.Time  	`json:"date"`
	CreatedAt   time.Time  	`json:"created_at"`
	UpdatedAt   time.Time  	`json:"updated_at"`
}

File: frontend/lib/models/expense.dart
class Expense {
  final int id;
  final double amount;
  final String category;
  final String description;
  final DateTime date;

  Expense({
    required this.id,
    required this.amount,
    required this.category,
    required this.description,
    required this.date,
  });

  factory Expense.fromJson(Map<String, dynamic> json) {
    return Expense(
      id: json['id'],
      amount: (json['amount'] as num).toDouble(),
      category: json['category'] ?? '',
      description: json['description'] ?? '',
      date: DateTime.parse(json['date']),
    );
  }
}
